tests/test_crypto.py

```diff
@@ -1740,7 +1740,7 @@ def test_construction(self):
         certificate = X509()
         assert isinstance(certificate, X509)
         assert type(certificate).__name__ == "X509"
-        assert type(certificate) == X509
+        assert type(certificate) is X509
```

Comment:
Are these identity check changes related to the PR?

Reply:
Not really - I added those to get the PR to pass all the tests. Should I separate these things out into a separate PR?

Reply:
Normally, yes — it's a good idea to submit atomic changes as it's harder to reason about things when a PR title/description refers to the change in a single line and the diff is flooded with an unproportional amount of arbitrary changes that don't support its stated purpose. This affects the cognitive complexity of the patch as the mental burden for the reader/reviewer gets higher.

However, the maintainers may still decide to accept such PRs on occasion. Especially since the activities aren't happening as often in this repo. And ultimately it's up to them.

Though, what I'm talking about is always applicable in general — smaller patches require less time to accept, because they are easier to understand and when there's more changes, a problem with parts of them would block/postpone the bits that would be otherwise acceptable.

Reply:
I submitted a separate PR for these fixes and that's now been approved. Do I need to scrap this PR and create a new one? Or can this one now be accepted anyway?

Reply:
You can perform a rebase and that'll make the diff here smaller.

Reply:
I did the rebase but was a bit of a mess. Somehow my branch I rebased still says it is one commit behind the origin.

Reply:
I synced the PR with all commits now so it should be good to go now

Reply:
> I did the rebase but was a bit of a mess. Somehow my branch I rebased still says it is one commit behind the origin.

That usually happens when you forget to `git fetch --all`. And rebasing should be done against the remote branch (the local one with the same name tends to be outdate).

Reply:
Once fetched, if your pointer to this upstream repository is called `origin`, use `git rebase origin/main`. If you use your fork as origin, then you may want to add a separate remote pointing to this repo too. You can do something like `git remote add upstream git@github.com:pyca/pyopenssl.git` and `git fetch upstream`. Then your rebase command would be `git rebase upstream/main` (assuming you're on the local `jan23-2024` branch when running it). If you're having troubles with empty commits, it's possible to clean them up with an interactive rebase, I suppose.